$NetBSD$

Use lib as libdir, following pkgsrc norms (and perhaps norms for all
but Linux).

Not yet filed upstream.

--- src/Makefile.defs.orig	2020-03-16 14:17:27.000000000 +0000
+++ src/Makefile.defs
@@ -210,9 +210,9 @@ endif
 
 endif
 
-ifeq ($(LIBSSL_SET_MUTEX_SHARED), 1)
+ifeq ($(LIBSSL_SET_MUTEX_SHARED), 2)
 CC_PMUTEX_OPTS = -pthread -DKSR_PTHREAD_MUTEX_SHARED
-LD_PMUTEX_OPTS = -pthread -rdynamic -ldl -Wl,-Bsymbolic-functions
+LD_PMUTEX_OPTS = -pthread -rdynamic -Wl,-Bsymbolic-functions
 else
 CC_PMUTEX_OPTS =
 LD_PMUTEX_OPTS =
@@ -443,15 +443,7 @@ endif
 
 $(info target architecture <$(ARCH)>, host architecture <$(HOST_ARCH)>)
 
-LIBDIR ?=
-ifeq ($(LIBDIR),)
-ARCHBSZ= $(shell echo $(ARCH) | sed -e 's/.*64.*/64b/')
-ifeq ($(ARCHBSZ),64b)
-	LIBDIR = lib64
-else
-	LIBDIR = lib
-endif
-endif
+LIBDIR = lib
 
 # dirs
 cfg_dir = etc/$(MAIN_NAME)/
@@ -1821,7 +1813,7 @@ ifeq ($(YACC),)
 endif
 YACC_FLAGS=-d -b core/cfg
 # on solaris add -lxnet (e.g. LIBS= -lxnet)
-LIBS= -ldl -lresolv
+LIBS= -lresolv
 LIB_PREFIX:=lib
 LIB_SUFFIX:=.so
 
@@ -2042,6 +2034,7 @@ ifeq ($(OS), netbsd)
 	ifneq ($(found_lock_method), yes)
 		C_DEFS+= -DUSE_SYSV_SEM  # try pthread sems
 		found_lock_method=yes
+		LIBS+= -pthread   #dlopen is in libc
 	endif
 	# check for ver >= 2.0.0
 	ifeq ($(shell [ $(OSREL_N) -ge 2000000 ] && echo has_kqueue), has_kqueue)
